dask/bag/core.py

```diff
@@ -341,20 +341,25 @@ def to_textfiles(self, path, name_function=str):
         return to_textfiles(self, path, name_function)
 
     def fold(self, binop, combine=None, initial=None):
-        """ Splittable reduction
+        """ Parallelizable reduction
 
```

Comment:
If this is confusing (rightly so) then lets go whole hog here.

Fold is like the builtin function `reduce` except that it works in parallel.  Fold takes two binary operator functions, one to reduce each partition of our dataset and another to combine results between partitions

1.  ``binop``: Binary operator to reduce within each partition
2.  ``combine``:  Binary operator to combine results from binop

Sequentially this would look like the following:

    intermediates = [reduce(binop, part) for part in partitions]
    final = reduce(combine, intermediates)

If only one function is given then it is used for both functions ``binop`` and ``combine`` as in the following example to compute the sum

    >>> add = lambda x, y: x + y
    >>> b.fold(add).compute()
    10

In full form we provide both binary operators as well as their default arguments

    >>> b.fold(binop=add, combine=add, initial=0).compute()
    10
    

Thoughts?